pkg/pipeline/sink/uploader/s3.go

```diff
@@ -40,18 +41,48 @@ const (
 	getBucketLocationRegion = "us-east-1"
 )
 
-// CustomRetryer wraps the SDK's built in DefaultRetryer adding additional
+// S3Retryer wraps the SDK's built in DefaultRetryer adding additional
 // custom features. Namely, to always retry.
-type CustomRetryer struct {
+type S3Retryer struct {
 	client.DefaultRetryer
 }
 
 // ShouldRetry overrides the SDK's built in DefaultRetryer because the PUTs for segments/playlists are always idempotent
-func (r CustomRetryer) ShouldRetry(_ *request.Request) bool {
+func (r S3Retryer) ShouldRetry(_ *request.Request) bool {
 	return true
 }
 
+// S3Logger only logs aws messages on upload failure
+type S3Logger struct {
+	mu   sync.Mutex
+	msgs []string
+}
+
+func (l *S3Logger) Log(args ...interface{}) {
+	msg := "aws sdk:"
+	for range len(args) {
+		msg += " %v"
+	}
+
+	l.mu.Lock()
+	if len(l.msgs) >= 10 {
+		l.msgs = append(l.msgs[1:], msg)
```

Comment:
If my understanding of go slices is correct, this will cause the array backing to slice to grow indefinitely until the logger gets destroyed as the beginning of the array will never be reclaimed. We may want to either copy the leftover part of the slice to the beginning of the array, or use ring buffer implementation?